gateway/proxy_eth.go

```diff
@@ -592,6 +607,7 @@ func (gw *Node) EthUnsubscribe(ctx context.Context, id ethtypes.EthSubscriptionI
 
 	ok, err := gw.target.EthUnsubscribe(ctx, id)
 	if err != nil {
+		// don't delete the subscription, it's "stuck" so should still count towards the limit
```

Comment:
chasing these calls down into eth.go, I think the only case here is that of "filter not found"; and in the case of EthSubscribe specifically, there's not really even a case where we get that

BUT there is one important caveat - the backend node is different to the one where the filter was originally installed, either because the node was restarted (in which case the error shouldn't count toward the limit in the gateway for the user), or because we have a problem with sticky sessions for this user - some reverse proxy is involved that has shunted this request to the wrong gateway or from the gateway to the wrong node (that latter one is a bit weird, but not impossible with a funky setup).

In the case of non-sticky sessions, then the gateway operator might be in a bit of trouble. Now we have a logger in here maybe it'd be worth printing a warning here.

Reply:
added log